realm/realm-library/src/androidTest/java/io/realm/OrderedCollectionChangeSetTests.java

```diff
@@ -0,0 +1,357 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm;
+
+import android.support.test.runner.AndroidJUnit4;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.concurrent.CountDownLatch;
+
+import io.realm.entities.AllTypes;
+import io.realm.rule.RunInLooperThread;
+import io.realm.rule.RunTestInLooperThread;
+import io.realm.rule.TestRealmConfigurationFactory;
+
+import static junit.framework.Assert.assertEquals;
+import static junit.framework.Assert.assertNotNull;
+import static junit.framework.Assert.assertNull;
+import static junit.framework.Assert.assertSame;
+import static junit.framework.Assert.fail;
+import static org.junit.Assert.assertArrayEquals;
+
+// Tests for the ordered collection fine grained notifications.
+// This should be expanded to test the notifications for RealmList as well in the future.
+@RunWith(AndroidJUnit4.class)
+public class OrderedCollectionChangeSetTests {
+
+    @Rule
+    public final TestRealmConfigurationFactory configFactory = new TestRealmConfigurationFactory();
+    @Rule
+    public final RunInLooperThread looperThread = new RunInLooperThread();
+
+    @Before
+    public void setUp() {
+    }
+
+    @After
+    public void tearDown() {
+    }
+
+    private void populateData(Realm realm, int testSize) {
+        realm.beginTransaction();
+        for (int i = 0; i < testSize; i++) {
+            realm.createObject(AllTypes.class).setColumnLong(i);
+        }
+        realm.commitTransaction();
+    }
+
+    // The args should be [startIndex1, length1, startIndex2, length2, ...]
+    private void checkRanges(OrderedCollectionChangeSet.Range[] ranges, long... indexAndLen) {
+        if ((indexAndLen.length % 2 != 0))  {
+            fail("The 'indexAndLen' array length is not an even number.");
+        }
+        if (ranges.length != indexAndLen.length / 2) {
+            fail("The lengths of 'ranges' and 'indexAndLen' don't match.");
+        }
+        for (int i = 0; i < ranges.length; i++) {
+            OrderedCollectionChangeSet.Range range = ranges[i];
+            long startIndex = indexAndLen[i * 2];
+            long length = indexAndLen[i * 2 + 1];
+            if (range.startIndex != startIndex || range.length != length) {
+                fail("Range at index " + i + " doesn't match start index " + startIndex + " length " + length + ".");
+            }
+        }
+    }
+
+    // Deletes AllTypes objects which's columnLong is in the indices array.
+    private void deleteObjects(Realm realm, long... indices) {
```

Comment:
It is actually indices. `RealmList` related tests will share the same test, and to `RealmList` it is indices.